### PR TITLE
Document of custom cipher for Docker Compose

### PR DESCRIPTION
**Pull request by overbid**
Custom cipher for discard warning and more security.

### CODE REVIEW

**Comment by kylemanna** [comment]
Why complicate it and create a variable that's only used once?

**Comment by overbid** [comment]
Same as $CIPHER from https://github.com/kylemanna/docker-openvpn/blob/master/docs/otp.md

File: docs/docker-compose.md [hunk 1]
```diff
@@ -22,6 +22,10 @@ services:
 
 ```bash
 docker-compose run --rm openvpn ovpn_genconfig -u udp://VPN.SERVERNAME.COM
+# with custom chiper for discard warning and more security like AES-128-GCM
```
Comment (pieterlange) [inline docs/docker-compose.md]: typo

File: docs/docker-compose.md [hunk 2]
```diff
@@ -22,6 +22,10 @@ services:
...
+export CIPHERNAME="your_cipher_name"
+docker-compose run --rm openvpn ovpn_genconfig -u udp://VPN.SERVERNAME.COM --cipther=$CIPHERNAME
```
Comment (pieterlange) [inline docs/docker-compose.md]: typo

**Comment by pieterlange** [comment]
Doesn't seem to add much value. Better to aim for secure defaults.

**Comment by buchdag** [comment]
@pieterlange > so, should we submit a PR to pass `--cipher AES-256-CBC` to openvpn by default and add `cipher AES-256-CBC` to the .ovpn files generated by ovpn_getclient ?

**Comment by breznak** [comment]
> so, should we submit a PR to pass --cipher AES-256-CBC to openvpn by default 

:+1: That would also resolve #163 
Is the version of openvpn with support for aes-256-cbc released as stable already? 

If people are running into issues (security vs usability), I'd vote for creating 2 conf files: `common.conf` and `-hardened.conf` to fit both needs.

**Comment by kylemanna** [comment]
> If people are running into issues (security vs usability), I'd vote for creating 2 conf files: common.conf and -hardened.conf to fit both needs.

People worried about security should be following the `docs/paranoid.md` document which uses AES-256 as an example.  The defaults for the repository are focused on a combination of safety and interoperability.

**Comment by kylemanna** [comment]
> Is the version of openvpn with support for aes-256-cbc released as stable already?

I think it has been for some time. Currently:

```
docker run --rm -it kylemanna/openvpn openvpn --show-ciphers
Unable to find image 'kylemanna/openvpn:latest' locally
latest: Pulling from kylemanna/openvpn
627beaf3eaaf: Already exists 
77d8000e7044: Pull complete 
fd416e3a2989: Pull complete 
db2ded8fd3a9: Pull complete 
Digest: sha256:9b10153d746ea893885d5348986d248278b17ad1e02ab8e74395f10b1692de76
Status: Downloaded newer image for kylemanna/openvpn:latest
The following ciphers and cipher modes are available
for use with OpenVPN.  Each cipher shown below may be
used as a parameter to the --cipher option.  The default
key size is shown as well as whether or not it can be
changed with the --keysize directive.  Using a CBC mode
is recommended. In static key mode only CBC mode is allowed.

AES-128-CBC  (128 bit key, 128 bit block)
AES-128-CFB  (128 bit key, 128 bit block, TLS client/server mode only)
AES-128-CFB1  (128 bit key, 128 bit block, TLS client/server mode only)
AES-128-CFB8  (128 bit key, 128 bit block, TLS client/server mode only)
AES-128-OFB  (128 bit key, 128 bit block, TLS client/server mode only)
AES-192-CBC  (192 bit key, 128 bit block)
AES-192-CFB  (192 bit key, 128 bit block, TLS client/server mode only)
AES-192-CFB1  (192 bit key, 128 bit block, TLS client/server mode only)
AES-192-CFB8  (192 bit key, 128 bit block, TLS client/server mode only)
AES-192-OFB  (192 bit key, 128 bit block, TLS client/server mode only)
AES-256-CBC  (256 bit key, 128 bit block)
AES-256-CFB  (256 bit key, 128 bit block, TLS client/server mode only)
AES-256-CFB1  (256 bit key, 128 bit block, TLS client/server mode only)
AES-256-CFB8  (256 bit key, 128 bit block, TLS client/server mode only)
AES-256-OFB  (256 bit key, 128 bit block, TLS client/server mode only)
CAMELLIA-128-CBC  (128 bit key, 128 bit block)
CAMELLIA-128-CFB  (128 bit key, 128 bit block, TLS client/server mode only)
CAMELLIA-128-CFB1  (128 bit key, 128 bit block, TLS client/server mode only)
CAMELLIA-128-CFB8  (128 bit key, 128 bit block, TLS client/server mode only)
CAMELLIA-128-OFB  (128 bit key, 128 bit block, TLS client/server mode only)
CAMELLIA-192-CBC  (192 bit key, 128 bit block)
CAMELLIA-192-CFB  (192 bit key, 128 bit block, TLS client/server mode only)
CAMELLIA-192-CFB1  (192 bit key, 128 bit block, TLS client/server mode only)
CAMELLIA-192-CFB8  (192 bit key, 128 bit block, TLS client/server mode only)
CAMELLIA-192-OFB  (192 bit key, 128 bit block, TLS client/server mode only)
CAMELLIA-256-CBC  (256 bit key, 128 bit block)
CAMELLIA-256-CFB  (256 bit key, 128 bit block, TLS client/server mode only)
CAMELLIA-256-CFB1  (256 bit key, 128 bit block, TLS client/server mode only)
CAMELLIA-256-CFB8  (256 bit key, 128 bit block, TLS client/server mode only)
CAMELLIA-256-OFB  (256 bit key, 128 bit block, TLS client/server mode only)

The following ciphers have a block size of less than 128 bits, 
and are therefore deprecated.  Do not use unless you have to.

BF-CBC  (128 bit key by default, 64 bit block)
BF-CFB  (128 bit key by default, 64 bit block, TLS client/server mode only)
BF-OFB  (128 bit key by default, 64 bit block, TLS client/server mode only)
CAST5-CBC  (128 bit key by default, 64 bit block)
CAST5-CFB  (128 bit key by default, 64 bit block, TLS client/server mode only)
CAST5-OFB  (128 bit key by default, 64 bit block, TLS client/server mode only)
DES-CBC  (64 bit key, 64 bit block)
DES-CFB  (64 bit key, 64 bit block, TLS client/server mode only)
DES-CFB1  (64 bit key, 64 bit block, TLS client/server mode only)
DES-CFB8  (64 bit key, 64 bit block, TLS client/server mode only)
DES-EDE-CBC  (128 bit key, 64 bit block)
DES-EDE-CFB  (128 bit key, 64 bit block, TLS client/server mode only)
DES-EDE-OFB  (128 bit key, 64 bit block, TLS client/server mode only)
DES-EDE3-CBC  (192 bit key, 64 bit block)
DES-EDE3-CFB  (192 bit key, 64 bit block, TLS client/server mode only)
DES-EDE3-CFB1  (192 bit key, 64 bit block, TLS client/server mode only)
DES-EDE3-CFB8  (192 bit key, 64 bit block, TLS client/server mode only)
DES-EDE3-OFB  (192 bit key, 64 bit block, TLS client/server mode only)
DES-OFB  (64 bit key, 64 bit block, TLS client/server mode only)
DESX-CBC  (192 bit key, 64 bit block)
IDEA-CBC  (128 bit key, 64 bit block)
IDEA-CFB  (128 bit key, 64 bit block, TLS client/server mode only)
IDEA-OFB  (128 bit key, 64 bit block, TLS client/server mode only)
RC2-40-CBC  (40 bit key by default, 64 bit block)
RC2-64-CBC  (64 bit key by default, 64 bit block)
RC2-CBC  (128 bit key by default, 64 bit block)
RC2-CFB  (128 bit key by default, 64 bit block, TLS client/server mode only)
RC2-OFB  (128 bit key by default, 64 bit block, TLS client/server mode only)
gost89  (256 bit key, 8 bit block, TLS client/server mode only)
gost89-cnt  (256 bit key, 8 bit block, TLS client/server mode only)
```

**Comment by buchdag** [comment]
The additional environment variable might be used to both pass a non default cipher to `openvpn` and have `ovpn_getclient` add the correct cipher to the .ovpn file.

I use `--cipher AES-256-CBC` and I still manage to forget that I have to add `cipher AES-256-CBC` to the .ovpn files from time to time.

**Comment by buchdag** [comment]
Disregard my last two comments, this is already done if you follow instructions for [Crypto Hardening](https://github.com/kylemanna/docker-openvpn/blob/master/docs/paranoid.md#crypto-hardening).

**Comment by kylemanna** [comment]
The secure defaults on the OpenVPN 2.4.x are much better. My machines use `AES-256-GCM` by default doing nothing more then upgrading the docker-openvpn image to `latest` and ensuring the clients are using OpenVPN 2.4.x.

Closing issue.  Anyone wanting to override the defaults can review the paranoid doc.

**Pull request closed without merging**